gax-grpc/src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -29,348 +29,58 @@
  */
 package com.google.api.gax.grpc;
 
-import com.google.api.core.AbstractApiFuture;
 import com.google.api.core.ApiFuture;
 import com.google.api.core.BetaApi;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.util.concurrent.SettableFuture;
 import com.google.longrunning.Operation;
-import com.google.longrunning.OperationsClient;
-import com.google.protobuf.Any;
-import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.Message;
-import io.grpc.Status;
-import java.util.concurrent.Callable;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Executor;
-import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-import org.threeten.bp.Duration;
 
 /** An ApiFuture which polls a service through OperationsApi for the completion of an operation. */
 @BetaApi
-public final class OperationFuture<ResponseT extends Message> extends AbstractApiFuture<ResponseT> {
-  static final Duration DEFAULT_POLLING_INTERVAL = Duration.ofSeconds(1);
+public final class OperationFuture<ResponseT extends Message> implements ApiFuture<ResponseT> {
```

Comment:
You're missing something important: the metadata returned by the API can change over time - some metadata types have progress % for the operation. So, you actually need to issue a new call each time to get updated metadata.